hooks/post/30-upgrade.rb

```diff
@@ -59,11 +59,10 @@ def fail_and_exit(message)
   kafo.class.exit 1
 end
 
-upgrade_step :remove_legacy_mongo
-
 if app_value(:upgrade)
   if [0, 2].include?(@kafo.exit_code)
     upgrade_tasks if module_enabled?('foreman')
+    remove_legacy_mongo if module_enabled?('foreman')
```

Comment:
I'd expect this to be done when the `katello` or `foreman_proxy_content` module is enabled.

Reply:
You're also removing the `upgrade_step`. Is this intentional?